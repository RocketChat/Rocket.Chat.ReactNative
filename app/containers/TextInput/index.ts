export * from './TextInput';
export * from './FormTextInput';
export * from './ControlledFormTextInput';
