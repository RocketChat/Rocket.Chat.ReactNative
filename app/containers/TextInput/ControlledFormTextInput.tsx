import React from 'react';
import { Control, Controller } from 'react-hook-form';

import { FormTextInput, IRCTextInputProps } from './FormTextInput';

interface IControlledFormTextInputProps extends IRCTextInputProps {
	control: Control<any>;
	name: string;
}

export const ControlledFormTextInput = ({ control, name, ...props }: IControlledFormTextInputProps) => (
	<Controller
		control={control}
		name={name}
		render={({ field: { onChange, value } }) => <FormTextInput onChangeText={onChange} value={value} {...props} />}
	/>
);
