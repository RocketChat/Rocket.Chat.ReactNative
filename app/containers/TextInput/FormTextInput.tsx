import { BottomSheetTextInput } from '@gorhom/bottom-sheet';
import React, { useState } from 'react';
import { StyleProp, StyleSheet, Text, TextInput as RNTextInput, TextInputProps, TextStyle, View, ViewStyle } from 'react-native';
import Touchable from 'react-native-platform-touchable';

import { useTheme } from '../../theme';
import sharedStyles from '../../views/Styles';
import ActivityIndicator from '../ActivityIndicator';
import { CustomIcon, TIconsName } from '../CustomIcon';
import { TextInput } from './TextInput';

const styles = StyleSheet.create({
	error: {
		...sharedStyles.textAlignCenter,
		paddingTop: 5
	},
	inputContainer: {
		marginBottom: 10
	},
	label: {
		marginBottom: 10,
		fontSize: 14,
		...sharedStyles.textSemibold
	},
	input: {
		...sharedStyles.textRegular,
		height: 48,
		fontSize: 16,
		paddingHorizontal: 16,
		paddingVertical: 10,
		borderWidth: 1,
		borderRadius: 4
	},
	inputIconLeft: {
		paddingLeft: 45
	},
	inputIconRight: {
		paddingRight: 45
	},
	wrap: {
		position: 'relative',
		justifyContent: 'center'
	},
	iconContainer: {
		position: 'absolute'
	},
	iconLeft: {
		left: 12
	},
	iconRight: {
		right: 12
	}
});

export interface IRCTextInputProps extends TextInputProps {
	label?: string;
	error?: any;
	loading?: boolean;
	containerStyle?: StyleProp<ViewStyle>;
	inputStyle?: StyleProp<TextStyle>;
	inputRef?: React.Ref<RNTextInput>;
	iconLeft?: TIconsName;
	iconRight?: TIconsName;
	left?: JSX.Element;
	bottomSheet?: boolean;
	onClearInput?: () => void;
}

export const FormTextInput = ({
	label,
	error,
	loading,
	containerStyle,
	inputStyle,
	inputRef,
	iconLeft,
	iconRight,
	onClearInput,
	value,
	left,
	testID,
	secureTextEntry,
	bottomSheet,
	placeholder,
	...inputProps
}: IRCTextInputProps): React.ReactElement => {
	const { colors } = useTheme();
	const [showPassword, setShowPassword] = useState(false);
	const showClearInput = onClearInput && value && value.length > 0;
	const Input = bottomSheet ? BottomSheetTextInput : TextInput;
	return (
		<View style={[styles.inputContainer, containerStyle]}>
			{label ? (
				<Text style={[styles.label, { color: colors.titleText }, error?.error && { color: colors.dangerColor }]}>{label}</Text>
			) : null}

			<View style={styles.wrap}>
				<Input
					style={[
						styles.input,
						iconLeft && styles.inputIconLeft,
						(secureTextEntry || iconRight || showClearInput) && styles.inputIconRight,
						{
							backgroundColor: colors.backgroundColor,
							borderColor: colors.separatorColor,
							color: colors.titleText
						},
						error?.error && {
							color: colors.dangerColor,
							borderColor: colors.dangerColor
						},
						inputStyle
					]}
					// @ts-ignore ref error
					ref={inputRef}
					autoCorrect={false}
					autoCapitalize='none'
					underlineColorAndroid='transparent'
					secureTextEntry={secureTextEntry && !showPassword}
					testID={testID}
					accessibilityLabel={placeholder}
					placeholder={placeholder}
					value={value}
					{...inputProps}
				/>

				{iconLeft ? (
					<CustomIcon
						name={iconLeft}
						testID={testID ? `${testID}-icon-left` : undefined}
						size={20}
						color={colors.auxiliaryText}
						style={[styles.iconContainer, styles.iconLeft]}
					/>
				) : null}

				{showClearInput ? (
					<Touchable onPress={onClearInput} style={[styles.iconContainer, styles.iconRight]} testID='clear-text-input'>
						<CustomIcon name='input-clear' size={20} color={colors.auxiliaryTintColor} />
					</Touchable>
				) : null}

				{iconRight && !showClearInput ? (
					<CustomIcon
						name={iconRight}
						testID={testID ? `${testID}-icon-right` : undefined}
						size={20}
						color={colors.bodyText}
						style={[styles.iconContainer, styles.iconRight]}
					/>
				) : null}

				{secureTextEntry ? (
					<Touchable onPress={() => setShowPassword(!showPassword)} style={[styles.iconContainer, styles.iconRight]}>
						<CustomIcon
							name={showPassword ? 'unread-on-top' : 'unread-on-top-disabled'}
							testID={testID ? `${testID}-icon-password` : undefined}
							size={20}
							color={colors.auxiliaryText}
						/>
					</Touchable>
				) : null}

				{loading ? (
					<ActivityIndicator
						style={[styles.iconContainer, styles.iconRight]}
						color={colors.bodyText}
						testID={testID ? `${testID}-loading` : undefined}
					/>
				) : null}
				{left}
			</View>
			{error && error.reason ? <Text style={[styles.error, { color: colors.dangerColor }]}>{error.reason}</Text> : null}
		</View>
	);
};
