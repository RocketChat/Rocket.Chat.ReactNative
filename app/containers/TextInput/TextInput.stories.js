/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, import/extensions, react/prop-types */
import React from 'react';
import { storiesOf } from '@storybook/react-native';

import { View, StyleSheet } from 'react-native';
import FormTextInput from './FormTextInput';

const styles = StyleSheet.create({
	paddingHorizontal: {
		paddingHorizontal: 14
	}
});

const stories = storiesOf('Text Input', module);

const item = {
	name: 'Rocket.Chat',
	longText: 'https://open.rocket.chat/images/logo/android-chrome-512x512.png'
};

const theme = 'light';

stories.add('Short and Long Text', () => (
	<>
		<View style={styles.paddingHorizontal}>
			<FormTextInput label='Short Text' placeholder='placeholder' value={item.name} theme={theme} />

			<FormTextInput label='Long Text' placeholder='placeholder' value={item.longText} theme={theme} />
		</View>
	</>
));
