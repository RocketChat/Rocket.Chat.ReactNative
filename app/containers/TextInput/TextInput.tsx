import React from 'react';
import { I18nManager, StyleProp, StyleSheet, TextInput as RNTextInput, TextStyle } from 'react-native';

import { IRCTextInputProps } from './FormTextInput';
import { themes } from '../../lib/constants';
import { useTheme } from '../../theme';

const styles = StyleSheet.create({
	input: {
		...(I18nManager.isRTL ? { textAlign: 'right' } : { textAlign: 'auto' })
	}
});

export interface IThemedTextInput extends IRCTextInputProps {
	style: StyleProp<TextStyle>;
}

export const TextInput = React.forwardRef<RNTextInput, IThemedTextInput>(({ style, ...props }, ref) => {
	const { theme } = useTheme();
	return (
		<RNTextInput
			ref={ref}
			style={[{ color: themes[theme].titleText }, style, styles.input]}
			placeholderTextColor={themes[theme].auxiliaryText}
			keyboardAppearance={theme === 'light' ? 'light' : 'dark'}
			{...props}
		/>
	);
});
