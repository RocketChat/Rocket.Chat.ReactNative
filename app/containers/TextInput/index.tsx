import React from 'react';
import { I18nManager, StyleProp, StyleSheet, TextInput, TextStyle } from 'react-native';

import { IRCTextInputProps } from './FormTextInput';
import { themes } from '../../lib/constants';
import { TSupportedThemes } from '../../theme';

const styles = StyleSheet.create({
	input: {
		...(I18nManager.isRTL ? { textAlign: 'right' } : { textAlign: 'auto' })
	}
});

export interface IThemedTextInput extends IRCTextInputProps {
	style: StyleProp<TextStyle>;
	theme: TSupportedThemes;
}

const ThemedTextInput = React.forwardRef<TextInput, IThemedTextInput>(({ style, theme, ...props }, ref) => (
	<TextInput
		ref={ref}
		style={[{ color: themes[theme].titleText }, style, styles.input]}
		placeholderTextColor={themes[theme].auxiliaryText}
		keyboardAppearance={theme === 'light' ? 'light' : 'dark'}
		{...props}
	/>
));

export default ThemedTextInput;
