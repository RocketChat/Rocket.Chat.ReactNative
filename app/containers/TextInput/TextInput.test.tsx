import { generateSnapshots } from '../../../.rnstorybook/generateSnapshots';
import * as stories from './TextInput.stories';

generateSnapshots(stories);
