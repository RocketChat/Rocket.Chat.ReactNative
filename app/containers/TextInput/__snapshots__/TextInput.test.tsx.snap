// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Snapshots: Icons should match snapshot 1`] = `
<View
  style={
    {
      "paddingHorizontal": 14,
    }
  }
>
  <View>
    <View>
      <View
        accessibilityLabel="Right icon. . Error: . , Rocket.Chat."
        accessible={true}
        style={
          [
            {
              "gap": 4,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Right icon
           
        </Text>
        <View
          accessible={true}
          style={
            {
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardAppearance="light"
            placeholder="placeholder"
            placeholderTextColor="#9EA2A8"
            style={
              [
                {
                  "color": "#1F2329",
                },
                [
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "fontFamily": "Inter",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                    "paddingVertical": 14,
                    "textAlign": "left",
                  },
                  undefined,
                  {
                    "paddingRight": 45,
                  },
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#9EA2A8",
                    "color": "#1F2329",
                  },
                  {},
                  undefined,
                ],
                {
                  "textAlign": "auto",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="Rocket.Chat"
          />
          <Text
            accessible={false}
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#2F343D",
                  "fontSize": 40,
                },
                [
                  {
                    "lineHeight": 40,
                  },
                  [
                    {
                      "position": "absolute",
                    },
                    {
                      "right": 12,
                    },
                  ],
                ],
                {
                  "fontFamily": "custom",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View>
      <View
        accessibilityLabel="Left icon. . Error: . , https://open.rocket.chat/images/logo/android-chrome-512x512.png."
        accessible={true}
        style={
          [
            {
              "gap": 4,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Left icon
           
        </Text>
        <View
          accessible={true}
          style={
            {
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardAppearance="light"
            placeholder="placeholder"
            placeholderTextColor="#9EA2A8"
            style={
              [
                {
                  "color": "#1F2329",
                },
                [
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "fontFamily": "Inter",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                    "paddingVertical": 14,
                    "textAlign": "left",
                  },
                  {
                    "paddingLeft": 45,
                  },
                  {},
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#9EA2A8",
                    "color": "#1F2329",
                  },
                  {},
                  undefined,
                ],
                {
                  "textAlign": "auto",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="https://open.rocket.chat/images/logo/android-chrome-512x512.png"
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#6C727A",
                  "fontSize": 40,
                },
                [
                  {
                    "lineHeight": 40,
                  },
                  [
                    {
                      "position": "absolute",
                    },
                    {
                      "left": 12,
                    },
                  ],
                ],
                {
                  "fontFamily": "custom",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View>
      <View
        accessibilityLabel="Both icons. . Error: . , https://open.rocket.chat/images/logo/android-chrome-512x512.png."
        accessible={true}
        style={
          [
            {
              "gap": 4,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Both icons
           
        </Text>
        <View
          accessible={true}
          style={
            {
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardAppearance="light"
            placeholder="placeholder"
            placeholderTextColor="#9EA2A8"
            style={
              [
                {
                  "color": "#1F2329",
                },
                [
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "fontFamily": "Inter",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                    "paddingVertical": 14,
                    "textAlign": "left",
                  },
                  {
                    "paddingLeft": 45,
                  },
                  {
                    "paddingRight": 45,
                  },
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#9EA2A8",
                    "color": "#1F2329",
                  },
                  {},
                  undefined,
                ],
                {
                  "textAlign": "auto",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="https://open.rocket.chat/images/logo/android-chrome-512x512.png"
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#6C727A",
                  "fontSize": 40,
                },
                [
                  {
                    "lineHeight": 40,
                  },
                  [
                    {
                      "position": "absolute",
                    },
                    {
                      "left": 12,
                    },
                  ],
                ],
                {
                  "fontFamily": "custom",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
          <Text
            accessible={false}
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#2F343D",
                  "fontSize": 40,
                },
                [
                  {
                    "lineHeight": 40,
                  },
                  [
                    {
                      "position": "absolute",
                    },
                    {
                      "right": 12,
                    },
                  ],
                ],
                {
                  "fontFamily": "custom",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View>
      <View
        accessibilityLabel="Icon and touchable clear input. . Error: . , https://open.rocket.chat/images/logo/android-chrome-512x512.png."
        accessible={true}
        style={
          [
            {
              "gap": 4,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Icon and touchable clear input
           
        </Text>
        <View
          accessible={true}
          style={
            {
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardAppearance="light"
            placeholder="placeholder"
            placeholderTextColor="#9EA2A8"
            style={
              [
                {
                  "color": "#1F2329",
                },
                [
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "fontFamily": "Inter",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                    "paddingVertical": 14,
                    "textAlign": "left",
                  },
                  undefined,
                  {
                    "paddingRight": 45,
                  },
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#9EA2A8",
                    "color": "#1F2329",
                  },
                  {},
                  undefined,
                ],
                {
                  "textAlign": "auto",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="https://open.rocket.chat/images/logo/android-chrome-512x512.png"
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
                "position": "absolute",
                "right": 12,
              }
            }
            testID="clear-text-input"
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#2F343D",
                    "fontSize": 40,
                  },
                  [
                    {
                      "lineHeight": 40,
                    },
                    undefined,
                  ],
                  {
                    "fontFamily": "custom",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Story Snapshots: Loading should match snapshot 1`] = `
<View
  style={
    {
      "paddingHorizontal": 14,
    }
  }
>
  <View>
    <View>
      <View
        accessibilityLabel="Loading false. . Error: . , Rocket.Chat."
        accessible={true}
        style={
          [
            {
              "gap": 4,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Loading false
           
        </Text>
        <View
          accessible={true}
          style={
            {
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardAppearance="light"
            placeholder="placeholder"
            placeholderTextColor="#9EA2A8"
            style={
              [
                {
                  "color": "#1F2329",
                },
                [
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "fontFamily": "Inter",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                    "paddingVertical": 14,
                    "textAlign": "left",
                  },
                  undefined,
                  {},
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#9EA2A8",
                    "color": "#1F2329",
                  },
                  {},
                  undefined,
                ],
                {
                  "textAlign": "auto",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="Rocket.Chat"
          />
        </View>
      </View>
    </View>
  </View>
  <View>
    <View>
      <View
        accessibilityLabel="Loading true. . Error: . , Rocket.Chat."
        accessible={true}
        style={
          [
            {
              "gap": 4,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Loading true
           
        </Text>
        <View
          accessible={true}
          style={
            {
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardAppearance="light"
            placeholder="placeholder"
            placeholderTextColor="#9EA2A8"
            style={
              [
                {
                  "color": "#1F2329",
                },
                [
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "fontFamily": "Inter",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                    "paddingVertical": 14,
                    "textAlign": "left",
                  },
                  undefined,
                  {},
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#9EA2A8",
                    "color": "#1F2329",
                  },
                  {},
                  undefined,
                ],
                {
                  "textAlign": "auto",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="Rocket.Chat"
          />
          <ActivityIndicator
            color="#2F343D"
            style={
              [
                {
                  "position": "absolute",
                },
                {
                  "right": 12,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Story Snapshots: Multiline should match snapshot 1`] = `
<View
  style={
    {
      "paddingHorizontal": 14,
    }
  }
>
  <View>
    <View>
      <View
        accessibilityLabel="Multiline text. . Error: . , Rocket.Chat

https://open.rocket.chat/images/logo/android-chrome-512x512.png
."
        accessible={true}
        style={
          [
            {
              "gap": 4,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Multiline text
           
        </Text>
        <View
          accessible={true}
          style={
            {
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardAppearance="light"
            multiline={true}
            placeholder="placeholder"
            placeholderTextColor="#9EA2A8"
            style={
              [
                {
                  "color": "#1F2329",
                },
                [
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "fontFamily": "Inter",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                    "paddingVertical": 14,
                    "textAlign": "left",
                  },
                  undefined,
                  {},
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#9EA2A8",
                    "color": "#1F2329",
                  },
                  {},
                  undefined,
                ],
                {
                  "textAlign": "auto",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="Rocket.Chat

https://open.rocket.chat/images/logo/android-chrome-512x512.png
"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Story Snapshots: SecureTextEntry should match snapshot 1`] = `
<View
  style={
    {
      "paddingHorizontal": 14,
    }
  }
>
  <View>
    <View>
      <View
        accessibilityLabel="Secure text disabled. . Error: . , Rocket.Chat."
        accessible={true}
        style={
          [
            {
              "gap": 4,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Secure text disabled
           
        </Text>
        <View
          accessible={true}
          style={
            {
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardAppearance="light"
            placeholder="placeholder"
            placeholderTextColor="#9EA2A8"
            style={
              [
                {
                  "color": "#1F2329",
                },
                [
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "fontFamily": "Inter",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                    "paddingVertical": 14,
                    "textAlign": "left",
                  },
                  undefined,
                  {},
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#9EA2A8",
                    "color": "#1F2329",
                  },
                  {},
                  undefined,
                ],
                {
                  "textAlign": "auto",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="Rocket.Chat"
          />
        </View>
      </View>
    </View>
  </View>
  <View>
    <View>
      <View
        accessibilityLabel="Secure text enabled. . Error: . ."
        accessible={true}
        style={
          [
            {
              "gap": 4,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Secure text enabled
           
        </Text>
        <View
          accessible={true}
          style={
            {
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardAppearance="light"
            placeholder="placeholder"
            placeholderTextColor="#9EA2A8"
            secureTextEntry={true}
            style={
              [
                {
                  "color": "#1F2329",
                },
                [
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "fontFamily": "Inter",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                    "paddingVertical": 14,
                    "textAlign": "left",
                  },
                  undefined,
                  {
                    "paddingRight": 45,
                  },
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#9EA2A8",
                    "color": "#1F2329",
                  },
                  {},
                  undefined,
                ],
                {
                  "textAlign": "auto",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="Rocket.Chat"
          />
          <View
            style={
              [
                {
                  "position": "absolute",
                },
                {
                  "right": 12,
                },
              ]
            }
          >
            <View
              accessibilityLabel="Show Password"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#2F343D",
                      "fontSize": 40,
                    },
                    [
                      {
                        "lineHeight": 40,
                      },
                      undefined,
                    ],
                    {
                      "fontFamily": "custom",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Story Snapshots: ShortAndLong should match snapshot 1`] = `
<View
  style={
    {
      "paddingHorizontal": 14,
    }
  }
>
  <View>
    <View>
      <View
        accessibilityLabel="Short Text. . Error: . , Rocket.Chat."
        accessible={true}
        style={
          [
            {
              "gap": 4,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Short Text
           
        </Text>
        <View
          accessible={true}
          style={
            {
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardAppearance="light"
            placeholder="placeholder"
            placeholderTextColor="#9EA2A8"
            style={
              [
                {
                  "color": "#1F2329",
                },
                [
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "fontFamily": "Inter",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                    "paddingVertical": 14,
                    "textAlign": "left",
                  },
                  undefined,
                  {},
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#9EA2A8",
                    "color": "#1F2329",
                  },
                  {},
                  undefined,
                ],
                {
                  "textAlign": "auto",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="Rocket.Chat"
          />
        </View>
      </View>
    </View>
  </View>
  <View>
    <View>
      <View
        accessibilityLabel="Long Text. . Error: . , https://open.rocket.chat/images/logo/android-chrome-512x512.png."
        accessible={true}
        style={
          [
            {
              "gap": 4,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 22,
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Long Text
           
        </Text>
        <View
          accessible={true}
          style={
            {
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardAppearance="light"
            placeholder="placeholder"
            placeholderTextColor="#9EA2A8"
            style={
              [
                {
                  "color": "#1F2329",
                },
                [
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "fontFamily": "Inter",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                    "paddingVertical": 14,
                    "textAlign": "left",
                  },
                  undefined,
                  {},
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#9EA2A8",
                    "color": "#1F2329",
                  },
                  {},
                  undefined,
                ],
                {
                  "textAlign": "auto",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="https://open.rocket.chat/images/logo/android-chrome-512x512.png"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
